### ISSUE TITLE
Changing the parent of a node causes field to flash red for awhile

### ISSUE THREAD

**Issue by emenslin** [specify/specify7#4808]
**Describe the bug**
When you change the parent of any tree node the field will go from blue to red and then back to blue again (if it is a valid tree structure). This is likely some sort of performance issue so the lower the performance the longer it will flash red for 

**To Reproduce**

1. Go to data entry -> any tree (e.g. taxon) 
2. Add a parent of taxon (that is not the lowest rank) 
3. See it flash red even though the parent is valid

**Expected behavior**
The field should not flash red if you are entering a valid tree rank 

**Screenshots**

https://github.com/specify/specify7/assets/150940840/0aa580f2-880e-423e-97e8-24f0a27bfc92

Please, also fill out the following information manually:

- OS: Windows 11
- Browser: Chrome
- Specify 7 Version: edge
- Database Name: lsumzmammals9823
- Collection name: mammals
- User Name: Spadmin
[Specify 7 System Information - 2024-04-18T17_37_29.842Z.txt](https://github.com/specify/specify7/files/15028450/Specify.7.System.Information.-.2024-04-18T17_37_29.842Z.txt)

**Comment by melton-jason** [comment]
This was related to the changes in #4726, and was originally noted in a review by @lexiclevenger at https://github.com/specify/specify7/pull/4726#pullrequestreview-1990102746:

> * [x]  On a new Taxon record, set the Parent to be a rank which is not the lowest in the tree
> * [x]   Clear the Parent QueryComboBox (the tree rank should clear as well)
> * [x]   Set the Parent of the Taxon to be a node which is the lowest rank in the tree
> * [x]   A saveBlocker should be set on the Parent field which reports Bad Tree Structure and the Tree Rank picklist should become disabled
> * [x]   Change the parent to a Taxon in which the possible ranks do not include the current rank (definitionItem) of the new Taxon record. (See [Restore bad tree structure saveBlocker #4726 (review)](https://github.com/specify/specify7/pull/4726#pullrequestreview-1974967848))
> * [x]   The tree rank should be set accordingly to the lowest enforceable rank
> * [x]   On an existing tree record, ensure the form opens without unload protect being triggered (as in [Going to a record set triggers a form change #3259](https://github.com/specify/specify7/issues/3259))
> * [x]   The tree rank should not be overwritten with a new value, and should use the existing rank of the record as shown in the Tree Viewer (especially when there are multiple possible ranks the node could take, see [Improve setting of default Tree Ranks #4659 (comment)](https://github.com/specify/specify7/pull/4659#issue-2187680064))
> 
> Looks good! One note--the parent field flashes red when changing it to a lower rank as seen here. Not sure if that's supposed to happen:
> 
>  Screen.Recording.2024-04-09.at.3.33.39.PM.mp4

---

The problem is occurring because while the Rank for the tree is being fetched/determined after the Parent changes, the rank is set to a 'unknown' value (`undefined`). 
While this is occurring, the tree structure is invalid.

**Comment by emenslin** [comment]
I don't think this is a big issue since I doubt most people would run into it but when artifically slowed down this behavior is more apparent.

https://github.com/specify/specify7/assets/150940840/c6da295d-8c69-4b5e-bbf6-acd6c65aef96

In the video this is a valid tree structure so ideally there should be no 'bad tree structure' error

**Comment by emenslin** [comment]
Can recreate in edge (7.9.6)
